src/webgl/material.js

```diff
@@ -54,28 +59,74 @@ require('./p5.Texture');
  * @alt
  * zooming Mandelbrot set. a colorful, infinitely detailed fractal.
  */
-p5.prototype.loadShader = function(vertFilename, fragFilename) {
+/**
+ * @method loadShader
+ * @param  {String}         vertFilename
+ * @param  {function}       [callback]
+ * @param  {function}       [errorCallback]
+ * @return {Object}
+ */
+p5.prototype.loadShader = function() {
   p5._validateParameters('loadShader', arguments);
+  var vertFilename, fragFilename, callback, errorCallback;
+  for (var i = 0; i < arguments.length; i++) {
+    if (typeof arguments[i] === 'function') {
+      if (typeof callback === 'undefined') {
+        callback = arguments[i];
+      } else if (typeof errorCallback === 'undefined') {
+        errorCallback = arguments[i];
+      }
+    } else if (typeof arguments[i] === 'string') {
+      if (!vertFilename) {
+        vertFilename = arguments[i];
+      } else if (!fragFilename) {
+        fragFilename = arguments[i];
+      }
+    }
+  }
```

Comment:
@Zalastax because you need to support all of these valid ones:
```javascript
loadShader(vert);
loadShader(vert, successCallback);
loadShader(vert, successCallback, errorCallback);
loadShader(vert, frag);
loadShader(vert, frag, successCallback);
loadShader(vert, frag, successCallback, errorCallback);
```

And dealing with all of these with if statements is more effort, at least in my mind, when we already have the FES there to warn people of misuses such as yours, but I do see your point.